### PR TITLE
issue #106: azure vm terraform module

### LINKED ISSUE #106: Create a terraform modules that enable the creation of azure vms
A terraform module needs to be created so we can provision Azure VM's inside Azure

### PR DESCRIPTION
**Pull request by ThomasCardin**
- Closes #106 

The naming convention used inside the azure-vm module is based on what was currently done

### CODE REVIEW

**Comment by ThomasCardin** [comment]
The `.tfvars` is up to date with the lastest change from this PR

File: terraform/modules/azure-vm/bastion.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
+resource "azurerm_public_ip" "bastion_public_ip" {
+  name                = var.bastion_public_ip_name
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  allocation_method   = var.bastion_public_ip_allocation_method
+  sku                 = var.bastion_public_ip_sku
+}
```
Comment (SonOfLope) [inline terraform/modules/azure-vm/bastion.tf:7]:
```suggestion
resource "azurerm_public_ip" "bastion_public_ip" {
  name                = var.bastion_public_ip_name
  location            = azurerm_resource_group.rg.location
  resource_group_name = azurerm_resource_group.rg.name
  allocation_method   = var.bastion_public_ip_allocation_method
  sku                 = var.bastion_public_ip_sku
}
```

File: terraform/modules/azure-vm/bastion.tf [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+resource "azurerm_bastion_host" "bastion_host" {
+  name                = var.bastion_host_name
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
```
Comment (SonOfLope) [inline terraform/modules/azure-vm/bastion.tf:12]:
```suggestion
  location            = azurerm_resource_group.rg.location
  resource_group_name = azurerm_resource_group.rg.name
```

**Review by SonOfLope — CHANGES_REQUESTED** [review]

**Review by SonOfLope — CHANGES_REQUESTED** [review]
Following naming conventions from other folders I think terraform/modules/azure-vm/linux-vm.tf is supposed be named main.tf

File: terraform/modules/azure-vm/bastion.tf (hunk 1 above)
Comment (ThomasCardin) [reply terraform/modules/azure-vm/bastion.tf]: Done

File: terraform/modules/azure-vm/bastion.tf (hunk 2 above)
Comment (ThomasCardin) [reply terraform/modules/azure-vm/bastion.tf]: Done

**Comment by ThomasCardin** [comment]
Re putting this PR to draft. Missing the setup to create a microsoft VM

**Review by SonOfLope — APPROVED** [review]
LGTM

**Comment by rngadam** [comment]
Can we audit the bastion setup against best practices in a separate issue?

https://goteleport.com/blog/security-hardening-ssh-bastion-best-practices/

**Pull request merged**